highlighters/main/main-highlighter.zsh

```diff
@@ -442,16 +509,20 @@ _zsh_highlight_highlighter_main_paint()
     # Parse the sudo command line
     if (( ! in_redirection )); then
       if [[ $this_word == *':sudo_opt:'* ]]; then
-        case "$arg" in
+        if [[ -n $flags_with_argument ]] &&
+           [[ $arg == '-'[$flags_with_argument] ]]; then
```

Comment:
This will not correctly highlight `doas -nu phy1729 echo`. Not a regression.

Reply:
Also pasted arguments `doas -uphy1729`, right?

Reply:
`doas -uphy1729 echo` is highlights `echo` correctly since `-uphy1729` is not matched as a flag with argument.

Reply:
Right; that's actually tested: https://github.com/danielshahaf/zsh-syntax-highlighting/blob/d838a29df342e9c02f2d006e63a5b1eb5210cfd7/highlighters/main/test-data/precommand3.zsh#L30

Reply:
The problem with `doas -nu phy1729 echo` is that we can't currently distinguish a `-n` flag that takes no argument from a `-n` flag that takes a `u` argument followed by `phy1729 echo` as two positional arguments, since we only know which flag-letters take arguments.

We could do something like... *coding noises* 2c15b0e996ad.

Reply:
This works in master and is covered by test-data/precommand4.zsh.